content/apps/using-oauth2.md

```diff
@@ -0,0 +1,157 @@
+# Using OAuth 2.0 to connect an App
+
+The PIM uses OAuth 2.0 to manage the authorization of Apps.
+
+![App activation diagram](../img/apps/app-activation-sequence-diagram.png)
+
+
+## Credentials
+
+To begin, you must obtain valid OAuth 2.0 client credentials by registering your App on the
+[Akeneo Marketplace](https://marketplace.akeneo.com/node/add/extension).
+
+Then, when any user want to connect his PIM to your App, the authorization process will go through the following steps.
+
+::: info
+In the following examples, we will use those fictional urls:  
+`https://my-pim.cloud.akeneo.com` for the PIM  
+`https://my-app.example.com/oauth/activate` for your App Activation URL  
+`https://my-app.example.com/oauth/callback` for your App Callback URL
+::::
+
+## Activation URL
+
+Your App must expose an activation URL.  
+When the PIM user sees your App in his PIM marketplace, he can click on the "Connect" button
+and he will be redirected to this activation URL with the PIM URL in the query.
+
+```
+
+https://my-app.example.com/oauth/activate?pim_url=https%3A%2F%2Fmy-pim.cloud.akeneo.com
+```
+
+## Authorization Request
+
+The user landed on your Activation URL and when you are ready to do so, you must start the Authorization Request.
+Like any other OAuth 2.0 application, you simply have to redirect the user to the Authorization Server (the PIM)
+with the following parameters:
+
+- `response_type` (required, must always be "code")
+- `client_id` (required)
+- `redirect_uri` (optional)
```

Comment:
shouldn't it be required?

Reply:
good question aaaand houston, we have a small problem.
This is a part we've done wrong.

According to the spec, the redirect_uri should be optional and should be validated against a whitelist (or a single one) of callback urls.

We do none of the 2, I've already added a ticket in the backlog.